### PR TITLE
Box labels

### PR DESCRIPTION
**Pull request by p-galligan**
Add script for creating box labels.

### CODE REVIEW

**Comment by helrond** [comment]
We still need to do a couple of things:

1. Write the de-duplicated data to a file (TSV or CSV). Should probably also include a header row.
2. review `get_objects` and `get_title` methods. These are both basically one-line functions, so I'm not sure they need to be a method.
3. consider adding additional logic to the `get_id` method to handle multipart ids.
4. Consider some exception handling for AS connection in `__init__` method

**Comment by p-galligan** [comment]
@helrond Struggling with a concise way to check for ids in multiple fields, but otherwise script is currently working.

**Review by helrond — COMMENTED** [review]
I think we can remove these print statements.

File: archivessnake/get_labels.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    def get_id(self):
+        """
+        Gets and returns the id of a resource record.
+        """
+        id = self.resource.id_0
+        return id
+
+    def get_containers(self, obj):
+        """
+        Checks whether the object has a container instance and returns the top container link.
```
Comment (helrond) [inline archivessnake/get_labels.py]: This docstring needs to be updated

File: archivessnake/get_labels.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        for obj in self.resource.tree.walk:
+            if len(obj.instances):
+                parent = self.get_parent(obj)
+                container = self.get_containers(obj)
+                print(resource_title, resource_id, parent, container)
+                label_data.append([resource_title, resource_id, parent, container])
+        print("Raw data", len(label_data))
+        print("Deduplicated data", len(set(tuple(row) for row in label_data)))
+        label_data = set(tuple(row) for row in label_data)
+        self.make_csv(label_data)
```
Comment (helrond) [inline archivessnake/get_labels.py]: These could be combined into a single line

File: archivessnake/get_labels.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            writer = csv.writer(csvfile)
+            writer.writerow(['ResourceTitle', 'ResourceID', 'ParentTitle', 'Container'])
+            writer.writerows(label_data)
+
+    def get_id(self):
+        """
+        Gets and returns the id of a resource record.
+        """
+        id = self.resource.id_0
+        return id
```
Comment (helrond) [inline archivessnake/get_labels.py]: I would think something like:

```
number = []
for n in range(0,2):
  try:
    number.append(getattr(self.resource, "id_{}".format(2))
  except KeyError:
    break
":".join(number)
```
Not sure if that's catching the right exception, but something along those lines...

**Comment by p-galligan** [comment]
Ok, have not done any exception handling. Only potential current issue I see is that you cannot sort the sheet by box number because the box and number are in the same cell. Additionally, it writes the boxes out of order to the csv. I believe this is due to running set on the label_data.

**Comment by helrond** [comment]
@p-galligan check out [this tutorial](https://docs.python.org/3/howto/sorting.html) on sorting. You should still be able to sort even if the container type and indicator are in a single field.

**Comment by p-galligan** [comment]
@helrond Do you think this is ready to merge? Should I send it along to Emeline?

**Review by helrond — APPROVED** [review]
Let's merge it! 🚀

**Pull request merged**